LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -302,7 +302,61 @@ public void AssignInstructorsToClassrooms()
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            string action = "Register students for classes";
+            ReportHeader(actionName: action);
+
+            if (ClassroomDictionary.Count == 0)
+            {
+                ReportError(missingResource: "Classroom", failedAction: action);
+            }
+            else if (ActiveStudentList.Count == 0)
+            {
+                ReportError(missingResource: "Student", failedAction: action);
+            }
+            else
+            {
+                int maxCount = 0;
+                foreach (Classroom classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        maxCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+
+            ReportFooter(actionName: action);
+        }
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int studentCount = 0;
+
+            if (ActiveStudentList.Count < classroom.MaximumCapacity - classroom.ItemList.Count)
```

Comment:
This code is too chatty and is not very readable:
  if (ActiveStudentList.Count < classroom.MaximumCapacity - classroom.ItemList.Count)
            {
                studentCount = AttributeGenerator.random.Next((classroom.MaximumCapacity - classroom.ItemList.Count - ActiveStudentList.Count) + 1);
            }
            else
            {
                studentCount = AttributeGenerator.random.Next(classroom.MaximumCapacity - classroom.ItemList.Count);
            }
Use meaningful names for your local variable:
uint remainingPlaceCount = classroom.MaximumCapacity - classroom.ItemList.Count;
unit activeStudentCount = ActiveStudentList.Count;
Then you have to choose the minimum of this two values for your upperLimit in the random value generator.
If you get the value, then you don't need to write AttributeGenerator.random.Next() twice.

Reply:
Fixed.